### PR TITLE
Tiling nodes

### PR DESCRIPTION
**Pull request by HedgehogCode**
Created two new nodes:
* Tile Loop Start: Splits images into tiles of configured size. Also, adds overlap between the tiles if configured.
* Tile Loop End: Merges the tiles created by a tile loop start into an image of the previous size.

### CODE REVIEW

**Comment by gab1one** [comment]
Looks cool 😎 ! 
Can you please cleanup the WIP commits? 
If you don't have testflows yet please create them, then we can merge this 👍

File: org.knime.knip.base/src/org/knime/knip/base/nodes/util/tilelooper/TileIteratorLoopEndNodeFactory.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
+/*
+ * ------------------------------------------------------------------------
+ *
+ *  Copyright (C) 2003 - 2014
```
Comment (gab1one) [inline org.knime.knip.base/src/org/knime/knip/base/nodes/util/tilelooper/TileIteratorLoopEndNodeFactory.java]: wrong date here, it should be 2003 - 2017, check this for all files please

File: org.knime.knip.base/src/org/knime/knip/base/nodes/util/tilelooper/TileIteratorLoopEndNodeFactory.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!DOCTYPE knimeNode PUBLIC "-//UNIKN//DTD KNIME Node 2.0//EN" "http://www.knime.org/Node.dtd">
+<knimeNode icon="tileiteratorend.png" type="LoopEnd">
+	<name>Tile Loop End</name>
+
+	<shortDescription>
+		Collects tiles of Tile Loop Start and recombines them to images.
```
Comment (gab1one) [inline org.knime.knip.base/src/org/knime/knip/base/nodes/util/tilelooper/TileIteratorLoopEndNodeFactory.xml]: I think something like: "Collects tiles **from** the Tile Loop Start **node** .." would be better.

File: org.knime.knip.base/src/org/knime/knip/base/nodes/util/tilelooper/TileIteratorLoopEndNodeFactory.xml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	</shortDescription>
+
+	<fullDescription>
+		<intro>
+			Merges the tiles of a Tile Loop Start into the original images.
```
Comment (gab1one) [inline org.knime.knip.base/src/org/knime/knip/base/nodes/util/tilelooper/TileIteratorLoopEndNodeFactory.xml]: Should be the same as the short description, I think by merging these two sentences together we'll get the best result.

File: org.knime.knip.base/src/org/knime/knip/base/nodes/util/tilelooper/TileIteratorLoopEndNodeFactory.xml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+			Also removes the overlap.
```
Comment (gab1one) [inline org.knime.knip.base/src/org/knime/knip/base/nodes/util/tilelooper/TileIteratorLoopEndNodeFactory.xml]: "Also removes *all* overlap between the tiles." 
Maybe additionally explain how it is removed (just discarded?)

File: org.knime.knip.base/src/org/knime/knip/base/nodes/util/tilelooper/TileIteratorLoopEndNodeFactory.xml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+		</intro>
+
+		<option name="Image column to merge">
```
Comment (gab1one) [inline org.knime.knip.base/src/org/knime/knip/base/nodes/util/tilelooper/TileIteratorLoopEndNodeFactory.xml]: Maybe call this option something like "Tile(s) Column"?

File: org.knime.knip.base/src/org/knime/knip/base/nodes/util/tilelooper/TileIteratorLoopEndNodeFactory.xml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+			The column which contains the tiles of the images.
+		</option>
+	</fullDescription>
+
+	<ports>
+		<inPort index="0" name="Processed tiles">
+			Processed tiles.
```
Comment (gab1one) [inline org.knime.knip.base/src/org/knime/knip/base/nodes/util/tilelooper/TileIteratorLoopEndNodeFactory.xml]: Make this a full sentence

File: org.knime.knip.base/src/org/knime/knip/base/nodes/util/tilelooper/TileIteratorLoopEndNodeFactory.xml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+		</inPort>
+		<outPort index="0" name="Merged images">
+			Merged images.
```
Comment (gab1one) [inline org.knime.knip.base/src/org/knime/knip/base/nodes/util/tilelooper/TileIteratorLoopEndNodeFactory.xml]: Make this a full sentence

File: org.knime.knip.base/src/org/knime/knip/base/nodes/util/tilelooper/TileIteratorLoopStartNodeFactory.xml [hunk 8]
```diff
@@ -0,0 +1,50 @@
+<!DOCTYPE knimeNode PUBLIC "-//UNIKN//DTD KNIME Node 2.0//EN" "http://www.knime.org/Node.dtd">
+<knimeNode icon="tileiterator.png" type="LoopStart">
+	<name>Tile Loop Start</name>
+	<shortDescription>
+		Loops over tiles of equal size.
+	</shortDescription>
+
+	<fullDescription>
+		<intro>
+			Tile Loop, creates tiles of equal size of an input image.
```
Comment (gab1one) [inline org.knime.knip.base/src/org/knime/knip/base/nodes/util/tilelooper/TileIteratorLoopStartNodeFactory.xml]: The Tile Loop Start node ....

File: org.knime.knip.base/src/org/knime/knip/base/nodes/util/tilelooper/TileIteratorLoopStartNodeFactory.xml [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+			Only the chosen input images are processed. Other columns will removed from the table.
+			The loop is executed once per input image.
+			<p>
+			The image is expanded so it can be split into tiles of equal size.
```
Comment (gab1one) [inline org.knime.knip.base/src/org/knime/knip/base/nodes/util/tilelooper/TileIteratorLoopStartNodeFactory.xml]: Point to the Out of Bounds strategy setting here

File: org.knime.knip.base/src/org/knime/knip/base/nodes/util/tilelooper/TileIteratorLoopStartNodeFactory.xml [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+<knimeNode icon="tileiterator.png" type="LoopStart">
+	<name>Tile Loop Start</name>
+	<shortDescription>
+		Loops over tiles of equal size.
+	</shortDescription>
+
+	<fullDescription>
+		<intro>
+			Tile Loop, creates tiles of equal size of an input image.
+			Only the chosen input images are processed. Other columns will removed from the table.
```
Comment (gab1one) [inline org.knime.knip.base/src/org/knime/knip/base/nodes/util/tilelooper/TileIteratorLoopStartNodeFactory.xml]: Only the images in the chosen column

**Comment by HedgehogCode** [comment]
I just added the requested changes.

**Comment by gab1one** [comment]
[Tiling_loop_disturber tests.zip](https://github.com/knime-ip/knip/files/1374840/Tiling_loop_disturber.tests.zip)

**Comment by HedgehogCode** [comment]
What's the current problem with missing cells. For me, the workflow executes perfectly using my latest commit. Missing cells are still missing cells after the loop and no exception is thrown.
Or should missing cells be handled differently?

**Comment by gab1one** [comment]
Now everything works nicely 👍

**Pull request merged**